plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -513,6 +515,35 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
         return getTableMetadata(session, ((IcebergTableHandle) table).getSchemaTableName());
     }
 
+    @Override
+    public ConnectorTableSchema getTableSchema(ConnectorSession session, ConnectorTableHandle tableHandle)
```

Comment:
We don't need to implement this method, since we already have to load the table here, and thus the default implementation that delegates to `getTableMetadata()` is not any more expensive.

The refactoring to implement this method also makes this PR harder to read. It's difficult to see the logic change for the snapshot schema.

Reply:
Thank you @electrum  for raising this point.
Indeed, implementing `getTableSchema` is not really needed. I've modified `getTableMetadata` to use the `schema` transported via `IcebergTableHandle` for achieving the functionality of having the table schema corresponding to snapshot in snapshot queries.